src/workflows/airqo_etl_utils/ml_utils.py

```
@@ -691,85 +806,657 @@ def flag_rule_based_faults(df: pd.DataFrame) -> pd.DataFrame:
+
+    @staticmethod
+    def _load_existing_isolation_forest_metrics() -> Optional[Dict[str, float]]:
+        try:
+            artifact = FaultDetectionUtils._load_model_object(
+                source_file=FaultDetectionUtils._get_isolation_forest_model_path()
+            )
+            if isinstance(artifact, dict) and isinstance(artifact.get("metrics"), dict):
+                return artifact["metrics"]
+        except FileNotFoundError:
+            return None
+        except Exception as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)